packages/ui-extensions/src/surfaces/customer-account/components/Page.d.ts

```diff
@@ -12,15 +12,19 @@ export interface PageProps {
 
 declare global {
   interface HTMLElementTagNameMap {
-    ['s-page']: HTMLElement & PageProps;
+    ['s-page']: PageProps;
```

Comment:
I think we still need to extend `HTMLElement` here; otherwise, non preact extensions will lose a bunch of properties like `addEventListener()`.

Reply:
@simontaisne can you elaborate on this more? Wouldn't the same be true for button or am I missing something? https://github.com/shop/world/pull/56242/files#diff-846cda7003c339964ea0e81d58cb7c1223223ab6bb4c82bf1b0fb0e23cd46b8eR49

Reply:
So essentially:
- For vanilla JS, we're extending `HTMLElement` to retain essential functionality like `addEventListener()` or `textContent`, even if some native HTML attributes don't apply. We'll aim to tidy this up later, as it's a lower priority.
- For the Preact version, we're moving away from extending HTMLElement to prevent confusion from TypeScript attribute suggestions.